src/planscape/plan/views.py

```diff
@@ -505,23 +507,37 @@ def create_scenario(request: HttpRequest) -> HttpResponse:
     except Exception as e:
         return HttpResponseBadRequest("Ill-formed request: " + str(e))
 
+
+def _serialize_scenario(scenario: Scenario, weights: QuerySet) -> dict:
+    result = ScenarioSerializer(scenario).data
+    result['priorities'] = {}
+
+    for weight in weights:
+        serialized_weight = ScenarioWeightedPrioritySerializer(weight).data
+        if 'priority' in serialized_weight and 'weight' in serialized_weight:
+            result['priorities'][Condition.objects.get(
+                pk=serialized_weight['priority']).condition_dataset.condition_name] = serialized_weight['weight']
+    return result
+
+
 @csrf_exempt
 def get_scenario(request: HttpRequest) -> HttpResponse:
     try:
         assert isinstance(request.GET['id'], str)
         scenario_id = request.GET.get('id', "0")
-        scenario_exists = Scenario.objects.get(id=scenario_id)
+        scenario = Scenario.objects.get(id=scenario_id)
 
         user = get_user(request)
 
-        if scenario_exists.owner != user:
+        if scenario.owner != user:
             raise ValueError(
                 "You do not have permission to view this scenario.")
-
-        return JsonResponse(ScenarioSerializer(scenario_exists).data, safe=False)
+        weights = ScenarioWeightedPriority.objects.select_related().filter(scenario=scenario)
+        return JsonResponse(_serialize_scenario(scenario, weights), safe=False)
     except Exception as e:
         return HttpResponseBadRequest("Ill-formed request: " + str(e))
-    
+
+
 @csrf_exempt
 def list_scenarios_for_plan(request: HttpRequest) -> HttpResponse:
     try:
```

Comment:
line 544 seems repetitive - maybe this is out of scope for this pr, but are there plans to replace this section with get_plan_by_id?

Reply:
I prefer to specify plan_id rather than just id in the requests where its asking for scenarios for a plan or something similar. This way the required id type is clear. Leaving as is for now and can address when all these changes are in